### PR TITLE
terminal: allow the find widget to work in detached terminals

### PR DESCRIPTION
**Pull request by connor4312**
- Abstracts some methods to an `IAnyTerminalInstance` interface that
  is common between attached and detached terminal instances.
- Some minor changes in the find contribution to make it work with
  detached instances after that.
- Make the find widget track the `activeFindWidget` since commands
  previously depended on the active terminal, which is not tracked for
	detached terminals (the alternative would be making the active
	terminal instance tracker also track detached terminals)

And then it 'just works' in the test results terminal

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminal.ts [hunk 1]
```diff
@@ -149,15 +149,66 @@ export interface IDetachedXTermOptions {
 	processInfo: ITerminalProcessInfo;
 }
 
+/**
+ * A generic interface implemented in both the {@link ITerminalInstance} (an
+ * interface used for terminals attached to the terminal panel or editor) and
+ * {@link IDetachedTerminalInstance} (a terminal used elsewhere in VS Code UI).
+ */
+export interface IAnyTerminalInstance {
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminal.ts:157]: Preferred name: `IBaseTerminalInstance`

File: src/vs/workbench/contrib/terminalContrib/find/browser/terminal.find.contribution.ts [hunk 2]
```diff
@@ -12,19 +12,25 @@ import { ContextKeyExpr } from 'vs/platform/contextkey/common/contextkey';
...
 
 class TerminalFindContribution extends Disposable implements ITerminalContribution {
 	static readonly ID = 'terminal.find';
 
-	static get(instance: ITerminalInstance): TerminalFindContribution | null {
+	/**
+	 * Currently focused find widget. This is used to track action context since
+	 * 'active terminals' are only tracked for non-detached terminal instanecs.
+	 */
+	static activeFindWidget: TerminalFindContribution | null = null;
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminalContrib/find/browser/terminal.find.contribution.ts:31]: pref: `static activeFindWidget?: TerminalFindContribution`

File: src/vs/workbench/contrib/terminalContrib/find/browser/terminal.find.contribution.ts [hunk 3]
```diff
@@ -174,23 +196,24 @@ registerActiveInstanceAction({
...
+		const widget = contr?.findWidget;
 		if (widget) {
 			widget.show();
 			widget.find(false);
 		}
 	}
 });
 
-registerActiveInstanceAction({
+registerActiveXtermAction({
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminalContrib/find/browser/terminal.find.contribution.ts:209]: Was this rename an accident?

**Review by Tyriar — CHANGES_REQUESTED** [review]
Looks good overall, just a few small comments

File: src/vs/workbench/contrib/terminalContrib/find/browser/terminal.find.contribution.ts (hunk 3 above)
Comment (connor4312) [reply src/vs/workbench/contrib/terminalContrib/find/browser/terminal.find.contribution.ts:209]: `registerActiveXtermAction` is a variation that works for both detached and attached terminals. `registerActiveInstanceAction` only applies to the attached `ITerminalInstance`

**Review by Tyriar — APPROVED** [review]

**Pull request merged**